imi/src/main/java/org/motechproject/nms/imi/service/impl/CsrValidatorServiceImpl.java

```diff
@@ -135,15 +142,26 @@ private void validateLanguageLocationCode(String code) {
     }
 
 
+    private Subscription validateSubscription(String id) {
+        Subscription sub = subscriptionService.getSubscription(id);
+        if (sub == null) {
+            String error = String.format("Subscription %s does not exist in the database", id);
+            LOGGER.warn(error);
+            String externalID = "Subscription" + (StringUtils.isBlank(id) ? "" : " " + id);
+            alertService.create(externalID, "CSR Validation", error, AlertType.HIGH, AlertStatus.NEW, 0, null);
+        }
+        return sub;
+    }
+
+
     //todo: IT
     @Override
     public void validateSummaryRecord(CallSummaryRecordDto record) {
-        Subscription sub = subscriptionService.getSubscription(record.getRequestId().getSubscriptionId());
+        Subscription sub = validateSubscription(record.getRequestId().getSubscriptionId());
```

Comment:
can we rename this to validateSummaryRecordDto? The overload on the type seems a little confusing.